### ISSUE TITLE
support multiple mirror tasks in same job with same source repository name

### ISSUE THREAD

**Issue by thebeautiful** [swellaby/vsts-mirror-git-repository#51]
I'm trying to mirror 2 separate git-repo's in one stage, one job, two tasks. Both git-repo's have the same original name, but live in 2 different organizations. Destination names are different of course.
I'm getting an error "fatal: destination path `xyz` already exists and is not an empty directory".

Workaround: spreading the two tasks on 2 different jobs keeps it apart.

**Comment by calebcartwright** [comment]
hey @thebeautiful, thanks for reaching out. The behavior you are experiencing is "expected" with the current version, because of the way the source repo is cloned into the local workspace. 

One enhancement I could see that would support your scenario would be a new input field in the task that would allow you to specify the repo directory for the clone. 

Assuming we called the new input something like `sourceGitRepositoryDirectory`:
```yml
- task: swellaby.mirror-git-repository.mirror-git-repository-vsts-task.mirror-git-repository-vsts-task@1
  displayName: 'Mirror Git Repository'
  inputs:
    sourceGitRepositoryUri: 'https://github.com/org1/foo.git'
    sourceGitRepositoryDirectory: 'foo-1',
    sourceVerifySSLCertificate: false
    destinationGitRepositoryUri: 'https://dev.azure.com/myorg/myproject/_git/mirror1'
    destinationGitRepositoryPersonalAccessToken: '$(destVar)'

- task: swellaby.mirror-git-repository.mirror-git-repository-vsts-task.mirror-git-repository-vsts-task@1
  displayName: 'Mirror Git Repository'
  inputs:
    sourceGitRepositoryUri: 'https://github.com/second-org/foo.git'
    sourceGitRepositoryDirectory: 'foo-2',
    sourceVerifySSLCertificate: false
    destinationGitRepositoryUri: 'https://dev.azure.com/myorg/myproject/_git/mirror2'
    destinationGitRepositoryPersonalAccessToken: '$(destVar)'
```

cc @traviskosarek - thoughts?

**Comment by thebeautiful** [comment]
Thanks for the response. If we are working in a temporary directory we might consider adding a random string to the dir name.  In that way you save the user the hassle of adding an extra parameter. 


Cheers,
Geerten

**Comment by calebcartwright** [comment]
> If we are working in a temporary directory we might consider adding a random string to the dir name.

Yeah that'd work too. I'd definitely consider a PR that solved this with that approach, though I'd personally prefer the task input. 

The task is really just a lightweight wrapper around git commands (in this case `git clone --mirror source-repo-url.git`). As folks know, git's default behavior is to clone into a new directory with the "user-friendly" name portion of the repo, but obviously allows the user to specify the directory as well, so IMHO it makes sense for the task to support that as well.

I'd prefer that we mirror (no pun intended 😄) git's behavior and have the user be responsible for explicitly determining the local directory to clone the source repo if they want/need to override gits default. Putting the responsibility on the task to override that behavior adds additional concerns (determining sufficient "randomness", avoiding file path length limit issues on Windows, etc.) that we want to avoid.

**Comment by calebcartwright** [comment]
cc @beverts312 as well if you have any opinions on this one

**Comment by beverts312** [comment]
Tough to say on this one, my initial instinct is go random because I think it is more _edge casey_ to have a reason to care about the name of the directory than to potentially run into a naming conflict, but I also think that scenario is pretty _edge casey_

**Comment by calebcartwright** [comment]
> but I also think that scenario is pretty edge casey

Completely agree and that's another reason I don't want really want to update the task to clone the repo into a random directory name by default. I generally don't like implementing a solution for an edge case that changes the behavior for all cases.

I'd rather add a new, optional, input field that folks in such an edge case could leverage, exactly like git does

**Comment by calebcartwright** [comment]
@thebeautiful - we've released version 1.1.11 which includes an optional new field you can use to override the directory the source repo is cloned into. 

We tested this out to confirm you can have two mirror tasks with two identically named source repos in the same job. 

If you have any other questions or run into any issues please let us know, and thanks for reaching out with your use case!
